src/blockchain/compact_filters/nakamoto.rs

```diff
@@ -0,0 +1,677 @@
+                                    Some(tx_details.txid)
+                                } else {
+                                    None
+                                }
+                            } else {
+                                None
+                            }
+                        })
+                        .collect::<HashSet<_>>();
+
+                    for txid in db_txs {
+                        unconfirm_tx(database, &txid)?;
+                    }
+                }
+                Event::BlockMatched {
+                    hash: _,
+                    header,
+                    height,
+                    transactions,
+                } => {
+                    let timestamp = header.time;
+                    for tx in transactions {
+                        // Check that we haven't processed this transaction before
+                        if database
+                            .iter_txs(false)?
+                            .iter()
+                            .find(|detail| {
+                                detail.txid == tx.txid()
+                                    && detail.confirmation_time.as_ref().map(|bt| bt.height)
+                                        == Some(height as u32)
+                            })
+                            .is_none()
+                        {
+                            debug!(
+                                "Processing Confirmed Transaction : {} at Block : {}",
+                                tx.txid(),
+                                height
+                            );
+                            add_tx(
+                                database,
+                                &tx,
+                                Some(height as u32),
+                                Some(timestamp as u64),
+                                &mut internal_max_deriv,
+                                &mut external_max_deriv,
+                            )?;
+                        }
+                    }
+                }
+                Event::FeeEstimated { height, fees, .. } => {
+                    self.add_fee_data(height as u32, fees);
+                }
+                Event::TxStatusChanged { txid, status } => {
+                    match status {
+                        TxStatus::Unconfirmed => {
+                            debug!("txid:{}, status : Broadcasted", txid);
+                        }
+                        TxStatus::Acknowledged { peer } => {
+                            debug!("txid: {}, status : ACK by {}", txid, peer);
+                        }
+                        TxStatus::Confirmed { height, .. } => {
+                            debug!("txid: {}, status : Confirmed at {}", txid, height);
+                        }
+                        TxStatus::Reverted => {
+                            debug!("Transaction reverted due to reorg: txid : {}", txid);
+                            let _ = unconfirm_tx(database, &txid)?;
```

Comment:
FYI, nakamoto will re-announce reverted transactions. I don't think we fire another event for them other than this one (but we could).

Reply:
I think this transaction reversion event should be sufficient to catch them..